lib/elastomer/client/delete_by_query.rb

```diff
@@ -105,16 +126,27 @@ def accumulate(item)
       # Returns a Hash of statistics about the bulk operation
       def execute
         ops = Enumerator.new do |yielder|
-          @client.scan(@query, @params.merge(:_source => false)).each_document do |hit|
+          @client.scan(@query, search_params).each_document do |hit|
             yielder.yield([:delete, hit.select { |key, _| ["_index", "_type", "_id", "_routing"].include?(key) }])
           end
         end
 
-        stats = @client.bulk_stream_items(ops, @params) { |item| accumulate(item) }
+        stats = @client.bulk_stream_items(ops, bulk_params) { |item| accumulate(item) }
         @response_stats["took"] = stats["took"]
         @response_stats
       end
 
+      # Remove parameters that are not valid for the _search endpoint
+      def search_params
+        params = @params.merge(:_source => false)
+        params.select {|p, _| SEARCH_PARAMETERS.include? p}
+      end
+
+      # Remove parameters that are not valid for the _bulk endpoint
+      def bulk_params
+        @params.clone.tap { |p| p.delete(:q) }
```

Comment:
We have ActiveSupport so you could use `Hash#except` here.

Reply:
Thank you for letting me know! I'll change it right away.

Reply:
Oops, I should have checked this more closely. Right now ActiveSupport is only a development dependency:

```
  spec.add_development_dependency "activesupport",      ">= 3.0"
```

I think you should leave it as-is. Sorry for not taking the time to check this out more closely.